pkg/util/utils.go

```diff
@@ -291,3 +293,31 @@ func ContainsString(slice []string, s string) bool {
 	}
 	return false
 }
+
+// ValidateUser validates minikube is run by the recommended user (privileged or regular)
```

Comment:
I like the idea, but let's find a better home for it with a more appropriate function signature. What do you think about moving this to pkg/drivers/drivers.go so that the call can be `driver.ValidatePermissions()?

Since this function will no longer be in the `main` package, it should return an error instead of exiting.

Reply:
I have moved this to the `drivers` file. I was wondering, do we have any middleware setup? With this type of implementation, I really don't like how in some places, I need to check the commandline arguments and in some places, what is loaded from the config file. If we don't have it, I think it would be best to have a middleware setup which can validate config/permissions after  arguments are saved to config file and we validate the config file only.

That way, in future, it would be easier to add more checks like `validatePermissions`. What do you think?

Reply:
Gotcha! No strong feelings about it. Although I think `pkg/drivers` should be drivers only, not using minikube code.

@tstromberg thoughts?